homeassistant/components/modbus/base_platform.py

```diff
@@ -68,7 +68,6 @@ def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
         self._value = None
         self._available = True
         self._scan_interval = int(entry[CONF_SCAN_INTERVAL])
-        self._available = self._scan_interval == 0
```

Comment:
This doesn't modify the scan interval. It compares the scan interval with 0 and stores the corresponding boolean value as `self._available`.

Reply:
Very good catch !! Sorry my bad, I have had a rough day.